streams/piping/pipe-through.any.js

```diff
@@ -56,167 +57,97 @@ promise_test(() => {
-
-test(() => {
-  const pairs = [
-    {},
-    { readable: undefined, writable: undefined },
-    { readable: 'readable' },
-    { readable: 'readable', writable: undefined },
-    { writable: 'writable' },
-    { readable: undefined, writable: 'writable' }
-  ];
-  for (let i = 0; i < pairs.length; ++i) {
-    const pair = pairs[i];
-    const rs = new ReadableStream();
-    assert_throws(new TypeError(), () => rs.pipeThrough(pair),
-                  `pipeThrough should throw for argument ${JSON.stringify(pair)} (index ${i});`);
-  }
-}, 'undefined readable or writable arguments should cause pipeThrough to throw');
-
-test(() => {
-  const invalidArguments = [null, 0, NaN, '', [], {}, false, () => {}];
-  for (const arg of invalidArguments) {
-    const rs = new ReadableStream();
-    assert_equals(arg, rs.pipeThrough({ writable: new WritableStream(), readable: arg }),
-                  'pipeThrough() should not throw for readable: ' + JSON.stringify(arg));
-    const rs2 = new ReadableStream();
-    assert_equals(rs2, rs.pipeThrough({ writable: arg, readable: rs2 }),
-                  'pipeThrough() should not throw for writable: ' + JSON.stringify(arg));
   }
-}, 'invalid but not undefined arguments should not cause pipeThrough to throw');
-
-test(() => {
 
-  const thisValue = {
-    pipeTo() {
-      assert_unreached('pipeTo should not be called');
+  const brs = new BadReadableStream({
+    start(controller) {
+      controller.close();
     }
+  });
+  const readable = new ReadableStream();
+  const writable = new WritableStream();
+  const result = brs.pipeThrough({ readable, writable });
+
+  assert_false(calledPipeTo, 'the overridden pipeTo should not have been called');
+  assert_equals(result, readable, 'return value should be the passed readable property');
+}, 'pipeThrough should not call pipeTo on this');
+
+test(t => {
+  let calledFakePipeTo = false;
+  const realPipeTo = ReadableStream.prototype.pipeTo;
+  t.add_cleanup(() => {
+    ReadableStream.prototype.pipeTo = realPipeTo;
+  });
+  ReadableStream.prototype.pipeTo = () => {
+    calledFakePipeTo = true;
   };
+  const rs = new ReadableStream();
+  const readable = new ReadableStream();
+  const writable = new WritableStream();
+  const result = rs.pipeThrough({ readable, writable });
 
-  methodThrows(ReadableStream.prototype, 'pipeThrough', thisValue, [undefined, {}]);
-  methodThrows(ReadableStream.prototype, 'pipeThrough', thisValue, [null, {}]);
+  assert_false(calledFakePipeTo, 'the monkey-patched pipeTo should not have been called');
+  assert_equals(result, readable, 'return value should be the passed readable property');
 
-}, 'pipeThrough should throw when its first argument is not convertible to an object');
+}, 'pipeThrough should not call pipeTo on the ReadableStream prototype');
 
-test(() => {
+const badReadables = [null, undefined, 0, NaN, true, 'ReadableStream', Object.create(ReadableStream.prototype)];
```

Comment:
May make more sense to put this in brand-checks.js. Especially if we plan on eventually switching to idlharness.js, that makes it easier to nuke that file all at once.